test/unit/parser/format_parse_test.cpp

```diff
@@ -881,3 +881,38 @@ TEST_F(format_parse_test, executable_name)
     EXPECT_EQ(executable_name[0], "parser_test");
     EXPECT_EQ(executable_name[1], "build");
 }
+
+TEST_F(format_parse_test, same_object_flag_and_option)
```

Comment:
https://github.com/seqan/sharg-parser/blob/31d32074dd29f2c2cb3b6afcf71cbc06a8ab8d0f/include/sharg/detail/format_parse.hpp#L106-L127

Reply:
https://github.com/seqan/sharg-parser/blob/31d32074dd29f2c2cb3b6afcf71cbc06a8ab8d0f/include/sharg/detail/format_parse.hpp#L716-L719

Reply:
Maybe 
```cpp
value = value || flag_is_set(short_id) || flag_is_set(long_id);
```

I.e., only ever set from `false` to `true`, but never from `true` to `false`.

Reply:
It's a general "problem" that different flags, options, positional options can write to the same value.

But at least with options, the overwriting only happens when the option is given; the flag always overwrites.